types/react-router/index.d.ts

```diff
@@ -64,7 +64,7 @@ export interface RouteComponentProps<P> {
 
 export interface RouteProps {
   location?: H.Location;
-  component?: React.SFC<RouteComponentProps<any> | undefined> | React.ComponentClass<RouteComponentProps<any> | undefined>;
+  component?: React.SFC<RouteComponentProps<any> & any | undefined> | React.ComponentClass<RouteComponentProps<any> & any | undefined>;
```

Comment:
Yes you could change it to ```any``` and it would be the exact same thing as what you have now.

That being said, @dalgeb and @KostyaEsmukov have raised valid concerns. This change will undo the benefit of the existing strong typing. I'm not familiar with the library but it seems like there isn't enough agreement about this change. In general we try to avoid ```any```s unless they're truly necessary.